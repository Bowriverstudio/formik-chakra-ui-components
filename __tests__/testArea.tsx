import React from "react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";
import * as Yup from "yup";

import TextArea from "../textArea";
import TestingWrapper from "../utils/testingWrapper";

const validationSchema = Yup.object({
  termsOfService: Yup.boolean()
    .required("The terms and conditions must be accepted.")
    .oneOf([true], "The terms and conditions must be accepted."),
});

describe("Formik Chakra Ui Components", () => {
  test("TextArea", () => {
    const json = TestRenderer.create(
      <TestingWrapper
        initialValues={{ termsOfService: false }}
        validationSchema={validationSchema}
      >
        <TextArea name="termsOfService"></TextArea>
      </TestingWrapper>
    ).toJSON();
    // const head = (helmetContext as FilledContext).helmet;
    expect(json).toMatchSnapshot();
  });

  test("TextArea renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <TestingWrapper
        initialValues={{ termsOfService: false }}
        validationSchema={validationSchema}
      >
        <TextArea name="termsOfService"></TextArea>
      </TestingWrapper>,
      div
    );
    expect(ReactDOM.unmountComponentAtNode(div)).toBe(true);
  });
});
