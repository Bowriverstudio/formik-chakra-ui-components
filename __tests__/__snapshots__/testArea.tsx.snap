// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formik Chakra Ui Components TextArea 1`] = `
<div
  className="chakra-container css-1plwrig"
>
  <div>
    <form
      action="#"
      noValidate={true}
      onReset={[Function]}
      onSubmit={[Function]}
    >
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <label
          className="chakra-form__label css-1ig2arj"
          htmlFor="email"
          id="field-1-label"
        />
        <div
          className="chakra-input__group css-4302v8"
        >
          <textarea
            className="chakra-textarea css-0"
            disabled={false}
            id="termsOfService"
            name="termsOfService"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            readOnly={false}
            required={false}
            value={false}
          />
        </div>
      </div>
    </form>
  </div>
</div>
`;
